riscv-rt/CHANGELOG.md

```diff
@@ -31,6 +31,15 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 - Now, `_start_rust` jumps to `hal_main` instead of `main` directly. At linker level,
   `hal_main` maps to `main` if not defined. However, we now allow HALs to inject
   additional configuration code before jumping to the final user's `main` function.
+- Now, `a0` is preserved in `s1` during the startup process.
+
+### Removed
+
+- Removed usage of the stack before `_start_rust`. This was unsound, as in the `.init`
+  section RAM is still uninitialized.
+- Now, `a1` and `a2` are **not** preserved during the startup process. New documentation
```

Comment:
I don't think clobbering `a1` at startup is a good idea. For instance, if `riscv-rt` code is being run as a kernel-like SBI payload, `a1` stores the address location of the `DeviceTree`.

I think this is just a convention, so not a hard requirement. I'm not aware of any conventions requiring preservation of the `a2` register.

Reply:
In RVI, we can store `a0-a2` in `s1-s3`. We can even use `s0-s2` and initialize the frame pointer later, as proposed in the other PR. But im RVE we only have `s0-s1`, so we need to use `a4` and/or `a5`. Using `aX` registers from preservation is not part of the ABI, so we must document it to avoid misuse.

Here, while we are not preserving `a1-a2`, we clearly specify in the documentation for `_mp_hook` and `__preinit` that these registers are reserved and must not be used.

Let me know what you prefer and I'll try it.